cli/src/pcluster/utils.py

```diff
@@ -579,3 +579,19 @@ def retrieve_supported_regions():
             with open(pkg_resources.resource_filename(__name__, "/resources/supported-regions"), encoding="utf-8") as f:
                 retrieve_supported_regions.cache = f.read().split("\n")
     return retrieve_supported_regions.cache
+
+
+def get_service_principal(service_name: str, partition: str, region: str = None, regional: bool = False) -> str:
```

Comment:
This new function should be covered by a unit test

Reply:
Done